SingularityUI/app/components/groupDetail/GroupDetail.jsx

```diff
@@ -13,7 +14,7 @@ class GroupDetail extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      showRequestId: _.first(props.group.requestIds)
+      showRequestId: props.params.requestId || _.first(props.group.requestIds),
```

Comment:
component state isn't needed anymore, and you can get rid of `onSelect={this.handleRequestSelect}` on the Nav component